### ISSUE TITLE
How to update...forking? Getting you out of this mess, if you want...

### ISSUE THREAD

**Issue by ksdavidc** [shabegom/obsidian-scratchblocks#2]
Scratchblocks has been updated to 3.6.1. Is this something I should worry about, or is the package file getting the right version automatically? (Is that how it works? How do I find out?) Makes me wonder if this project is really something I should fork, since I got you in this mess. The problem is, I only partly understand what is going on. I have looked at the repo and I get things here and there, but not enough to really know what is going on. I am new to this kind of thing; HTML, CSS, basic JS, jQuery I can handle, but Obsidian's inner workings are entirely out of my wheelhouse, let alone all the commonJS ESM etc. going on, building packages and plugins voodoo. Bits and pieces relate to bits and pieces of things I have sort of done or seen done, but I wonder if it would help if I had a brief "mid-level" rundown of how this thing works so that I could imagine learning what I would need to take it off your hands…not that I am unhappy, just to do right by you.

**Comment by hayribakici** [comment]
@ksdavidc @shabegom I am willing to fork this repo and continue developing this plugin. My first PR #4 also adds display and language settings. I would also be interested in publishing it, since I also use obsidian to prepare my worksheets. Would that be something you'd be interested in?

**Comment by ksdavidc** [comment]
I CAN'T BELIEVE SOMEONE ELSE IS USING THIS PLUGIN!!! I would be willing to help out, within the limits of my capabilities, if you desire, but I would probably need to be brought up to speed or pointed in the right direction.

**Comment by ksdavidc** [comment]
I will add that after several efforts to regain contact with them via assorted means, it appears that @shabegom  is AWOL, and is unlikely to be monitoring this thread.

**Comment by hayribakici** [comment]
I am glad to help out. If @shabegom is AWOL for good, then we could use the fork and work from there. What do you think @ksdavidc ?

**Comment by ksdavidc** [comment]
Yes, I do think that would be appropriate, if you are willing and able. It's been about 3 years since I first tried to reach @shabegom. Shabegom in fact did not originally make this plugin for themself, but as a sort of random favor to me, a stranger, in answer to a question I had posed on a discord server. I think it was supposed to be a one off, and it still basically works, but then scratchblocks got updated and a few bugs or inconsistencies were discovered. 

This all started when I tried to naively use scratchblocks in obsidian using the js file (per the [instructions](https://github.com/scratchblocks/scratchblocks)), but I got stymied when I found out that "scratchblocks is ESM, but Obsidian is CommonJS". (Please understand I do not know what that sentence means except in a hand wavy way. I am hoping perhaps you know what this is?) At any rate, I tried to get help on discord and @shabegom took pity on me and figured out a workaround. But they did so in a way that is opaque to me, so I don't know how to update it. Someone said if they can't do it, no one can, and it became a challenge.

https://discord.com/channels/686053708261228577/875720842443649045/1012061798649839699 

At the time I knew little about either how scratch or obsidian worked under the hood. I know a little bit more about each now, but still not enough to do this without a guide, as I think it is a real hack, as neither is natively intended to work with the other (unless you count both being text based). I would be eager to learn what I had to to get a handle on this, but I don't know where to start and how to avoid rabbit holes.

(In point of fact, what I really want to do is the obverse process: develop a plugin for scratch that allows me to get the scratchblock code for code stacks via the scratch gui, e.g. via a right click...then I could paste that directly in obsidian...the circle of life! But blockly!!)

**Comment by hayribakici** [comment]
@ksdavidc That sounds like a good idea. I can imagine linking the scratch file, and you get the blocks from  the figures. 

However, for now I could merge the PR #4 into my fork and keep it as a Pull Request here. In case @shabegom would  like to merge it any time in the future. Or he gives ownership to this repo.

Besides from that, I would then publish the fork as a first version. From there, we then could discuss what could be done next.

**Comment by ksdavidc** [comment]
By all means. Assuming this is the fork https://github.com/hayribakici/obsidian-scratchblocks/, I've forked it and will start working through understanding it. It's been a while, so I am going to be learning as I go.  :-)

ps Sorry about the TMI, but if you've been drifting on a life raft in the ocean for 3 years, and a big tanker shows up to rescue you, it's hard not to mention how dangerous the sharks were, even if getting on the tanker is really the first priority.

**Comment by hayribakici** [comment]
all good :)

**Comment by hayribakici** [comment]
@ksdavidc Since there is no discussion area to post stuff like that, but I am happy to announce the first version of the fork of this plugin.

https://community.obsidian.md/plugins/scratchblocks

**Comment by ksdavidc** [comment]
Mazeltov! 🎉
